### ISSUE TITLE
DelegatingMap documention is misleading

### ISSUE THREAD

**Issue by jamesderlin** [dart-lang/collection#171]
The documentation for [`DelegatingMap.addAll`](https://pub.dev/documentation/collection/latest/collection/DelegatingMap/addAll.html) states:

> The operation is equivalent to doing `this[key] = value` for each key and associated value in other. 

That led me to mistakenly believe that a class that derives from `DelegatingMap` could override `operator []=` and change the behavior of `addAll` (and `addEntries`, etc.) for free.  Alas, no, dartdoc automatically copied that from the base `Map.addAll` documentation, and `addAll` simply calls `addAll` on the delegate. (This behavior makes `DelegatingMap` seem significantly less useful.)

The implementation shown in the `DelegatingMap.addAll` documentation also doesn't make this as obvious as it could be because the delegate is named `_base` instead of `_delegate`.

This probably applies to the other `Delegating...` classes from `package:collection`, but I haven't looked at them too carefully.

**Comment by lrhn** [comment]
Good point. The behavior is equivalent only to the default behavior of `[]=`. The documentation should be rewritten.

**Comment by jamesderlin** [comment]
I realize now that I probably should have started from `MapMixin` instead, which has the behavior I expected.  I get the `...Mixin` and `Delegating...` classes mixed up.

I'll also note that [the documentation for `ListMixin`](https://api.dart.dev/stable/2.19.2/dart-collection/ListMixin-class.html) mentions `DelegatingList`, but [the `DelegatingList` documentation](https://pub.dev/documentation/collection/latest/collection/DelegatingList-class.html) does not refer to `ListMixin`.  Meanwhile there are no references in either direction for `MapMixin`/`DelegatingMap` nor for `SetMixin`/`DelegatingSet`.  It'd be nice if each referenced the other and explained what the differences are.
